examples/visual-question-answering/run_pipeline.py

```diff
@@ -0,0 +1,142 @@
+
+
+def main():
+    parser = argparse.ArgumentParser()
+
+    parser.add_argument(
+        "--model_name_or_path",
+        default=None,
+        type=str,
+        help="Path to pre-trained model",
+    )
+    parser.add_argument(
+        "--image_path",
+        default=None,
+        type=str,
+        nargs="*",
+        help='Path to image as input. Can be a single string (eg: --image_path "URL1"), or a list of space-separated strings (eg: --image_path "URL1" "URL2")',
+    )
+    parser.add_argument(
+        "--topk",
+        default=1,
+        type=int,
+        help="topk num",
+    )
+    parser.add_argument(
+        "--question",
+        default=None,
+        type=str,
+        nargs="*",
+        help='question as input. Can be a single string (eg: --question "Q1"), or a list of space-separated strings (eg: --question "Q1" "Q2")',
+    )
+    parser.add_argument(
+        "--use_hpu_graphs",
+        action="store_true",
+        help="Whether to use HPU graphs or not. Using HPU graphs should give better latencies.",
+    )
+    parser.add_argument(
+        "--bf16",
+        action="store_true",
+        help="Whether to perform in bf16 precision.",
+    )
+    parser.add_argument("--batch_size", type=int, default=1, help="Input batch size.")
+    parser.add_argument("--warmup", type=int, default=3, help="Number of warmup iterations for benchmarking.")
+    parser.add_argument("--n_iterations", type=int, default=5, help="Number of inference iterations for benchmarking.")
+    args = parser.parse_args()
+
+    adapt_transformers_to_gaudi()
+    image_pathes = args.image_path
+    image_pathes_len = len(image_pathes)
+
+    if args.batch_size > image_pathes_len:
+        # Dynamically extends to support larger batch sizes
+        num_path_to_add = args.batch_size - image_pathes_len
+        for i in range(num_path_to_add):
+            image_pathes.append(image_pathes[i % image_pathes_len])
+    elif args.batch_size < image_pathes_len:
+        image_pathes = image_pathes[: args.batch_size]
+
+    questions = args.question
+    questions_len = len(questions)
+    if args.batch_size > questions_len:
+        # Dynamically extends to support larger batch sizes
+        num_question_to_add = args.batch_size - questions_len
+        for i in range(num_question_to_add):
+            questions.append(questions[i % questions_len])
+    elif args.batch_size < questions_len:
+        questions = questions[: args.batch_size]
+
+    images = []
+
+    for image_path in image_pathes:
+        images.append(PIL.Image.open(requests.get(image_path, stream=True, timeout=3000).raw).convert("RGB"))
+
+    if args.bf16:
+        model_dtype = torch.bfloat16
+    else:
+        model_dtype = torch.float32
+
+    generator = pipeline(
+        "visual-question-answering",
+        model=args.model_name_or_path,
+        torch_dtype=model_dtype,
+        device="hpu",
+    )
+    if not generator.model.can_generate() and args.use_hpu_graphs:
```

Comment:
Is there an issue if the model has a `generate` method?

Reply:
thanks, remove the constraints